intermediate_source/per_sample_grads.py

```diff
@@ -168,8 +168,23 @@ def compute_loss(params, buffers, sample, target):
 # we can double check that the results using ``grad`` and ``vmap`` match the
 # results of hand processing each one individually:
 
-for per_sample_grad, ft_per_sample_grad in zip(per_sample_grads, ft_per_sample_grads.values()):
-    assert torch.allclose(per_sample_grad, ft_per_sample_grad, atol=3e-3, rtol=1e-5)
+# Get the parameter names in the same order as per_sample_grads
+
+for name, ft_per_sample_grad in ft_per_sample_grads.items():
+    # Find the corresponding manually computed gradient
+    idx = list(model.named_parameters()).index((name, model.get_parameter(name)))
+    per_sample_grad = per_sample_grads[idx]
+
+    # Check if shapes match and reshape if needed
+    if per_sample_grad.shape != ft_per_sample_grad.shape and per_sample_grad.numel() == ft_per_sample_grad.numel():
+        ft_per_sample_grad = ft_per_sample_grad.view(per_sample_grad.shape)
```

Comment:
I'm a bit confused by this part

Is the issue that `torch.allclose` now fails due to the ordering of `per_sample_grad` and `ft_per_sample_grad`  being different during zip?

Reply:
cc @zou3519 do you know whether there were any changes to vmap/functorch that might have caused the numerics to change here?